packager/media/formats/mp2t/mpeg1_header.cc

```diff
@@ -0,0 +1,228 @@
+}
+
+size_t Mpeg1Header::GetMinFrameSize() const {
+  return kMpeg1HeaderMinSize + 1;
+}
+
+static inline size_t
+Mpeg1SampleRate(uint8_t sr_idx, int version)
+{
+  static int sr_version_indexes[] = {2, -1, 1, 0};
+  CHECK(version != 1);
+  DCHECK_LT(sr_idx, kMpeg1SampleRateTableSize);
+  return kMpeg1SampleRateTable[sr_idx][sr_version_indexes[version]];
+}
+
+static inline size_t
+Mpeg1BitRate(uint8_t btr_idx, int version, int layer)
+{
+  static int btr_version_indexes[] = {2, -1, 1, 0};
+  static int btr_layer_indexes[] = {-1, 2, 1, 0};
+
+  assert(version != 1);
+  assert(layer != 0);
+  int vidx = btr_version_indexes[version];
+  int lidx = btr_layer_indexes[layer];
+  if (vidx == 2)
+    vidx = 1;
+  if (vidx == 1 && lidx > 1)
+    lidx = 1;
+  assert(btr_idx < 15);
+  assert(vidx * 3 + lidx < 5);
+
+  DCHECK_LT(btr_idx, kMpeg1BitrateTableSize);
+  return kMpeg1BitrateTable[btr_idx][vidx*3+lidx] * 1000;
+}
+
+static inline size_t
+Mpeg1FrameSize(int layer, int bitrate, int sample_rate, uint8_t padded)
+{
+  RCHECK(sample_rate > 0);
+  if (layer == kMpeg1L_1)
+    return (12 * bitrate / sample_rate + padded) * 4;
+  return 144 * bitrate / sample_rate + padded;
+}
+
+size_t Mpeg1Header::GetSamplesPerFrame() const {
+  RCHECK((layer_ > 0) && (layer_ <= kMpeg1SamplesPerFrameTableSize));
+  return kMpeg1SamplesPerFrameTable[layer_ - 1];
+}
+
+bool Mpeg1Header::Parse(const uint8_t* mpeg1_frame, size_t mpeg1_frame_size) {
+  DCHECK(mpeg1_frame);
+
+  if (mpeg1_frame_size < kMpeg1HeaderMinSize)
+    return false;
+
+  BitReader frame(mpeg1_frame, mpeg1_frame_size);
+  // Verify frame starts with sync bits (0x7ff).
+  uint32_t sync;
+  RCHECK(frame.ReadBits(11, &sync));
+  RCHECK(sync == 0x7ff);
+  // MPEG version and layer.
+  RCHECK(frame.ReadBits(2, &version_));
+  RCHECK(version_ != kMpeg1V_INV);
+  RCHECK(frame.ReadBits(2, &layer_));
+  RCHECK(layer_ != kMpeg1L_INV);
+  RCHECK(frame.ReadBits(1, &protection_absent_));
+
+  uint8_t btr_idx;
+  RCHECK(frame.ReadBits(4, &btr_idx));
+  bitrate_ = Mpeg1BitRate(btr_idx, version_, layer_);
+
+  uint8_t sr_idx;
+  RCHECK(frame.ReadBits(2, &sr_idx));
+  sample_rate_ = Mpeg1SampleRate(sr_idx, version_);
+
+  RCHECK(frame.ReadBits(1, &padded_));
+  // Skip private stream bit.
+  RCHECK(frame.SkipBits(1));
+
+  RCHECK(frame.ReadBits(2, &channel_mode_));
+  // Skip Mode extension
+  RCHECK(frame.SkipBits(2));
+  // Skip copyright, origination and emphasis info.
+  RCHECK(frame.SkipBits(4));
+
+  return true;
+}
+
+size_t Mpeg1Header::GetHeaderSize() const {
+  // Unlike ADTS, for MP3, the whole frame is included in the media sample, so
+  // return 0 header size.
+  return 0;
+}
+
+size_t Mpeg1Header::GetFrameSize() const {
+  return Mpeg1FrameSize(layer_, bitrate_, sample_rate_, padded_);
+}
+
+size_t Mpeg1Header::GetFrameSizeWithoutParsing(const uint8_t* data,
+                                               size_t num_bytes) const {
+  DCHECK_GT(num_bytes, static_cast<size_t>(1));
```

Comment:
Shouldn't it be 2 instead of 1 as the code below requires at least more than 2 bytes.